### PR TITLE
added scoreboard and timer

### PR DESCRIPTION
**Pull request by willharrison**
was having trouble squashing these three commits, so I just left them.

### CODE REVIEW

**Comment by ALSchwalm** [comment]
This does not display correctly for me. I think you are generally going to have a hard time doing this with a custom font, as web font stuff seems to break all the time. Perhaps try to get this working with a font everyone will have (like Arial).

2015-01-31-192139_988x627_scrot

**Comment by willharrison** [comment]
ah, that's no good.  I will change it.  it looks just spacing tho.  does it look ok on level change?  I might leave it there.

**Comment by ALSchwalm** [comment]
Honestly, I'm not a fan of the font either way. If you really want to deal with a custom font, perhaps something like the "Quarterly" font, from the page you linked in the readme?

**Comment by willharrison** [comment]
eh, I tried that one and wasn't a fan.  I will just go back to arial.

**Comment by willharrison** [comment]
go try it now see if it works for you.

**Comment by ALSchwalm** [comment]
The dotted separator is still sticking out, and things don't look quite centered.

2015-01-31-195825_396x549_scrot

**Comment by willharrison** [comment]
yeah, they are off center because I am using the dotted separator to center them.  how would you fill about everything but the timer being left justified?

**Comment by ALSchwalm** [comment]
I don't have any strong feelings about it if that looks good. Have you tried setting the 'anchor' property of the text to {x: 0.5, y:0.5}? Then the position will be with respect to the center of the text.

**Comment by willharrison** [comment]
yeah, that helps.  try it now.

**Comment by ALSchwalm** [comment]
Seems to break again as soon as I clear a quad

**Comment by willharrison** [comment]
my bad, forgot to change the update string.  should work now.

**Comment by ALSchwalm** [comment]
The value of streak seems to not always correspond with the number that pops up. Is 'streak' counting the blocks that are cleared because they are floating?

**Comment by willharrison** [comment]
looks like it's not updating until the next batch in cleared.  what do you mean by floating?

**Comment by ALSchwalm** [comment]
You're right, it's just not updating at the right time.

**Comment by willharrison** [comment]
i'm an idiot.  it's working now.

**Comment by ALSchwalm** [comment]
Working for me now. Can you change 'streak' to something else though? Maybe 'best chain' or 'longest chain' something? Then squash these recent commits (I'm ok with the three you had), and we should be good.

**Comment by willharrison** [comment]
I'm having trouble squashing my "merge conflict" commits.

**Comment by ALSchwalm** [comment]
We can look at that tomorrow. For now, it looks like, "Best Chain" is hanging off the edge of the scoreboard. You should really make those lines separate text objects, and just use a smaller font size for that string. Spacing things with newlines is kind of a hack.

**Comment by willharrison** [comment]
I'll do that.

**Comment by thirtythreeforty** [comment]
@willharrison, Adam asked me to help with squashing this.  Since you're changing a published commit history anyway, I'd recommend rebasing the entire PR.  From your master branch, you want to rebase against Adam's master branch.  My usual scheme is that I add the forked repository as another remote called `upstream`:
```
$ git remote add upstream git@github.com:ALSchwalm/Quad.git
$ git fetch upstream
```
The branch `upstream/master` will appear.  Now you can, from your local `master`, do:
```
$ git rebase --interactive upstream/master
```
And edit the rebase script to look like
```
pick 58ba3c9 add scoreboard
pick ca3d81d timer
squash 97d08dc timer
squash c56317e bug
```
(I don't know exactly what you want to squash.)

You'll get merge conflicts, meaning the lines were changed in both revisions.  They look like
```
<<<<<<< HEAD
    <script data-main="app" src="js/lib/require.js"></script>
=======
    <link rel="stylesheet" type="text/css" href="assets/fonts/blox/stylesheet.css">
>>>>>>> 58ba3c9... add scoreboard
```
It only looks like you have two in this case, in `index.html` and `js/app/score.js`.  Merge the lines manually and carefully add the resolutions with `git add`.  Run `git rebase --continue` when you're done.  The rebase will continue (you may be stopped with more conflicts again).

At this point you have to force-push to the remote branch, which is the only way that I've worked out to keep the GitHub pull request open.  So:
```
$ git push -f
```
and you should be set.

Hope that helps.

**Comment by willharrison** [comment]
@thirtythreeforty that was very helpful, thanks.  It has been squashed into one commit now.

File: js/app/score.js [hunk 1]
```diff
@@ -19,28 +20,73 @@ function(config, music, background){
...
+
+        this.board.scorebackdrop = game.add.graphics(0, 0);
+        this.board.scorebackdrop.beginFill(0x000000, .5);
+        this.board.scorebackdrop.drawRect(offsets.x + 135, offsets.y + 155, 200, 50, 10);
+
+        this.board.chainbackdrop = game.add.graphics(0, 0);
+        this.board.chainbackdrop.beginFill(0x000000, .5);
+        this.board.chainbackdrop.drawRect(offsets.x + 135, offsets.y + 247, 200, 50, 10);
+
+        //var text = "\nLevel\n1\nScore\n0\nBest Chain\n0";
```
Comment (ALSchwalm) [inline js/app/score.js]: Please remove this

Comment (willharrison) [inline js/app/score.js]: the comment or all of it?

Comment (ALSchwalm) [inline js/app/score.js]: Just the commented bit

Comment (willharrison) [inline js/app/score.js]: it's removed

**Pull request merged**